### ISSUE TITLE
SelectCheckboxMenu: Keyboard navigation improvements

### ISSUE THREAD

**Issue by fcjm** [primefaces/primefaces#8456]
**Describe the defect**
Our application has recently been tested for accessibility. I am currently reviewing the components we use in terms of keyboard navigation. I think there can be some improvements made to the keyboard navigation of the SelectCheckboxMenu component to make it more accessible. I can also provide a PR for this. The problems I currently have with this component are the following:

- Focus is not reset to the trigger once the Overlay is closed
    - instead it gets set to the start of the page
- The user can tab out of the overlay on both ends (Shift+Tab/Tab)
    - since the overlay is located at the end of the page, the focus will be set there as well

**Reproducer**
The current behaviour can be tested on the showcase. 
https://www.primefaces.org/showcase/ui/input/checkboxMenu.xhtml?jfwid=b16ee

**Environment:**
 - PF Version: _11.0.0_
 - Affected browsers: _ALL_

**Expected behavior**
The focus should be set to the triggering element once the overlay is closed.
Inside the overlay the focus should be trapped.
